File: Workspace/Personal Training/JSW/Baekjoon/9012/solution.cs
using System.Text;

StringBuilder sb = new StringBuilder();
using (StreamReader sr = new StreamReader(Console.OpenStandardInput()))
{
    int T = int.Parse(sr.ReadLine());
    for (int i = 0; i < T; i++)
    {
        Stack<char> stack = new Stack<char>();
        bool isVPS = true;
        foreach (var c in sr.ReadLine())
        {
            if (c == '(') stack.Push('(');
            else
            {
                if (stack.Count == 0)
                {
                    isVPS = false;
                    break;
                }
                stack.Pop();
            }
        }
        if (stack.Count != 0) isVPS = false;
        sb.AppendLine(isVPS ? "YES" : "NO");
    }
}

Console.Write(sb);

File: Workspace/Personal Training/JSW/Baekjoon/9012/solution.py
T = int(input())
for _ in range(T):
    stack = []
    is_vps = True
    line = input()
    for c in line:
        if c == '(':
            stack.append(c)
        else:
            if not stack:
                is_vps = False
                break
            stack.pop()
    if stack:
        is_vps = False
    print('YES' if is_vps else 'NO')